python/mxnet/gluon/trainer.py

```diff
@@ -113,6 +113,36 @@ def _init_kvstore(self):
 
         self._kv_initialized = True
 
+
+    @property
+    def learning_rate(self):
+        return self._optimizer.lr
```

Comment:
This kind of implementation couples the two classes together (i.e. Trainer must know the structure of optimizer). Instead add accessors in optimizer and make reading learning rate the concern of optimizer.

Reply:
resolved